build-logic/src/main/kotlin/publishing/PublishingExtension.kt

```diff
@@ -0,0 +1,17 @@
+package publishing
+
+abstract class PublishingExtension {
+
+    var pom: Pom? = null
```

Comment:
it's a mess. you have both an option to configure via method and via this exposed property.  and to configure it one is required to do some imports(!).  

do something like this:

```kotlin
  internal lateinit var pom: Pom

  fun pom(
    block: Pom.() -> Unit
  ) {
    pom = Pom().apply(block)
  }
  
  data class Pom(var name: String = "", var description: String = "")
```

fields filling validations can be performed in plugin code or in pom method itself